include/eve/arch/wasm/as_register.hpp

```diff
@@ -0,0 +1,48 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/arch/wasm/predef.hpp>
+
+#include <type_traits>
+
+namespace eve
+{
+template<typename T> struct logical;
+struct wasm_128_;
+}
+
+#if defined(EVE_HW_WASM)
+
+namespace eve
+{
+// Check the types are fine then return *the* WASM SIMD type: v128_t
+template<typename Type, typename Size> struct as_register<Type, Size, eve::wasm_128_>
+{
+  static constexpr auto find()
+  {
+    constexpr auto width = sizeof(Type) * Size::value;
+    if constexpr( width <= 16 )
+    {
+      if constexpr( std::is_same_v<Type, double> ) return v128_t {};
+      else if constexpr( std::is_same_v<Type, float> ) return v128_t {};
+      else if constexpr( std::is_integral_v<Type> ) return v128_t {};
```

Comment:
Why if constexpr if  the result is always the same? To do the assert? Maybe then just use the `plain` concept?